src/validations/html/htmlInspector.js

```diff
@@ -38,8 +39,37 @@ const errorMap = {
       },
     };
   },
+  'invalid-list-children': (error) => {
+    const tag = error.context.tagName.toLowerCase();
+
+    return {
+      reason: 'invalid-list-children',
+      payload: {
+        tag,
+        children: 'li',
+      },
+    };
+  },
 };
 
+function invalidListChildrenValidator(listener, reporter) {
+  listener.on('element', (elementName, domElement) => {
+    if (domElement.childNodes.length &&
+        (elementName === 'ul' || elementName === 'ol')) {
```

Comment:
Project style for multiline if statements would be:

```js
if (
  domElement.childNodes.length &&
    (elementName === 'ul' || elementName === 'ol')
) {
```

Reply:
(I _really_ wish eslint had better support for enforcing consistency with that sort of thing)

Reply:
In other news, is it worth including `<dl>`s in here as well?

Reply:
It's weird that ESLint seems to support that style of spacing with functions, but not conditionals or loops.

I can modify the code to support an error about required children for the `<dl>`, since `<dl>` contains `<dt>` or `<dd>`- what do you think?

Were you thinking of the `<menu>` tag, which can also contain `<li>` elements?

Reply:
I hadn’t given any thought to the `<menu>` tag—a quick glance at MDN suggests that the rules for that one are considerably more complicated, but in the general case you can put more or less whatever you want in there, so I wouldn’t worry about it too much.

As for `<dl>`, seems like a nice-to-have, but my main observation was just that `<dl>`, like `<ul>` and `<ol>`, doesn’t permit text node children so might be an appropriate addition (even though the children it *does* permit are different)